src/test/java/com/puppycrawl/tools/checkstyle/api/FullIdentTest.java

```diff
@@ -151,6 +152,12 @@ public void testFullIdentArrayInit() throws Exception {
     }
 
     private static FullIdent prepareFullIdentWithCoordinates(int columnNo, int lineNo) {
+        final DetailAstImpl parent = new DetailAstImpl();
+        parent.setType(TokenTypes.TYPE);
+        parent.setColumnNo(1);
+        parent.setLineNo(1);
+        parent.setText("Parent");
+
```

Comment:
Real input would always have a parent here, I added this to avoid a NPE in `FullIdent.createFullIdent()`. It didn't make sense to add a null check to `appendBrackets` just to pass a UT.